ui/lib/homepage.dart

```diff
@@ -46,45 +70,81 @@ class Homepage extends StatelessWidget {
+                        textAlign: TextAlign.center,
+                        style: TextStyle(
+                          fontSize: 26,
+                          color: Colors.white,
+                        )
+                      )
+                    );
+                  }
+
+                  return Container(
+                    margin: const EdgeInsets.fromLTRB(0, 30, 0, 0),
+                    constraints: const BoxConstraints(maxWidth: 1100),
+                    alignment: Alignment.center,
+                    child: Column(
                       children: [
+                        FeaturedApp(
+                          id: apps[0].id,
+                          name: apps[0].name,
+                          backgroundUrl: apps[0].background,
+                          description: apps[0].description,
+                          onPlay: widget.onPlay
+                        ),
                         Container(
-                          margin: const EdgeInsets.all(20),
-                          child: const Text(
-                            'Available applications',
-                            style: TextStyle(
-                              fontSize: 26,
-                              color: Colors.white,
-                            ),
+                          margin: const EdgeInsets.fromLTRB(0, 60, 0, 0),
+                          child: Row(
+                            mainAxisAlignment: MainAxisAlignment.start,
+                            children: [
+                              Container(
+                                margin: const EdgeInsets.all(20),
+                                child: const Text(
+                                  'Available applications',
+                                  style: TextStyle(
+                                    fontSize: 26,
+                                    color: Colors.white,
+                                  ),
+                                ),
+                              ),
+                            ],
                           ),
                         ),
+                        AppList(apps: apps, onPlay: widget.onPlay)
                       ],
                     ),
-                  ),
-                  AppList(apps: apps, onPlay: onPlay)
-                ],
-              ),
-            ),
+                  );
+                } else if (snapshot.hasError) {
+                  return Container(
+                    margin: const EdgeInsets.fromLTRB(0, 250, 0, 0),
+                    alignment: Alignment.center,
+                    child: Text(
+                        '${snapshot.error}',
+                        textAlign: TextAlign.center,
+                        style: TextStyle(
+                          fontSize: 26,
+                          color: Colors.white,
+                        )
+                      )
+                    );
+                }
+                return const CircularProgressIndicator();
```

Comment:
The indicator appears on the top of the page but we should center it instead

Reply:
Fixed.